source/signaling_api.c

```diff
@@ -0,0 +1,1754 @@
+                        {
+                            /* Ignore following servers. */
+                            break;
+                        }
+                    }
+
+                    updateUris( &pIceServerList->iceServer[pIceServerList->iceServerNum], pair.value, pair.valueLength );
+                }
+                else if( strncmp( pair.key, "Username", pair.keyLength ) == 0 )
+                {
+                    if( pIceServerList->iceServer[pIceServerList->iceServerNum].pUserName != NULL )
+                    {
+                        pIceServerList->iceServerNum++;
+
+                        if( pIceServerList->iceServerNum >= SIGNALING_AWS_ICE_SERVER_MAX_NUM )
+                        {
+                            /* Ignore following servers. */
+                            break;
+                        }
+                    }
+
+                    pIceServerList->iceServer[pIceServerList->iceServerNum].pUserName = pair.value;
+                    pIceServerList->iceServer[pIceServerList->iceServerNum].userNameLength = pair.valueLength;
+                }
+                else {
+                    /* Skip unknown messages. */
+                }
+
+                jsonResult = JSON_Iterate( pIceSingleServerBuffer, iceSingleServerBufferLength, &iceSingleServerStart, &iceSingleServerNext, &pair );
+            }
+
+        }
+        else {
+            /* All parsed. */
+            pIceServerList->iceServerNum++;
+            break;
+        }
+    }
+
+    return result;
+}
+
+SignalingResult_t Signaling_Init( SignalingContext_t * pCtx,
+                                  SignalingAwsControlPlaneInfo_t * pAwsControlPlaneInfo )
+{
+    SignalingResult_t result = SIGNALING_RESULT_OK;
+    int length = 0;
+
+    /* input check */
+    if( ( pCtx == NULL ) || ( pAwsControlPlaneInfo == NULL ) )
+    {
+        result = SIGNALING_RESULT_BAD_PARAM;
+    }
+
+    /* Set region. */
+    if( result == SIGNALING_RESULT_OK )
+    {
+        memset( pCtx, 0, sizeof( SignalingContext_t ) );
+
+        if( pAwsControlPlaneInfo->pRegion == NULL )
+        {
+            length = snprintf( pCtx->region, SIGNALING_AWS_REGION_MAX_LENGTH, "%.*s",
+                               ( int ) strlen( SIGNALING_AWS_DEFAULT_REGION ), SIGNALING_AWS_DEFAULT_REGION );
+        }
+        else {
+            length = snprintf( pCtx->region, SIGNALING_AWS_REGION_MAX_LENGTH, "%.*s",
+                               ( int ) pAwsControlPlaneInfo->regionLength, pAwsControlPlaneInfo->pRegion );
+        }
+
+        if( length < 0 )   //LCOV_EXCL_BR_LINE
+        {
+            result = SIGNALING_RESULT_SNPRINTF_ERROR; // LCOV_EXCL_LINE
+        }
+        else if( length >= SIGNALING_AWS_CONTROL_PLANE_URL_MAX_LENGTH )
+        {
+            result = SIGNALING_RESULT_REGION_LENGTH_TOO_LARGE;
+        }
+        else {
+            pCtx->regionLength = length;
+        }
+    }

```

Comment:
0-length region check - user can pass in empty string?

Reply:
If it's 0, user would fail to connect with signaling server. That case is basically same as incorrect AWS region string.